src/linter/types.go

```diff
@@ -15,7 +15,7 @@ type warningString string
 
 type shapeTypeProp struct {
 	key   string
-	types []meta.Type
+	types []meta.PhpDocType
```

Comment:
Or is this the problem that you are trying to solve? Trying to determine whether or not the types come from the code or from PHPDoc?